Go/Go.sublime-syntax

```diff
@@ -1202,72 +1352,83 @@ contexts:
   pop-const-expressions:
-    - include: pop-on-semicolon
-    - include: match-comments
+    - include: pop-before-semicolon
     - match: (?=\S)
       set:
-        - include: pop-on-terminator
+        - include: pop-before-terminator
         - match: \biota\b
           scope: constant.language.go
         - include: match-any
 
   pop-var-type-and-or-assignment:
-    - include: pop-on-terminator
-    - include: match-comments
+    - include: pop-before-terminator
     - match: =
       scope: keyword.operator.assignment.go
       set: pop-var-expressions
     - match: (?=\S)
       set: [pop-var-assignment-or-terminate, pop-type]
 
   pop-var-assignment-or-terminate:
-    - include: pop-on-terminator
-    - include: match-comments
+    - include: pop-before-terminator
     - match: =
       scope: keyword.operator.assignment.go
       set: pop-var-expressions
     - include: pop-before-nonblank
 
   # Note: this doesn't support multiline expressions.
   pop-var-expressions:
-    - include: pop-on-semicolon
-    - include: match-comments
+    - include: pop-before-semicolon
     - match: (?=\S)
       set:
-        - include: pop-on-terminator
+        - include: pop-before-terminator
         - include: match-any
 
-  pop-type-assertion:
-    - match: \(
-      scope: punctuation.section.parens.begin.go
-      set:
-        - match: \)
-          scope: punctuation.section.parens.end.go
-          pop: true
-        - include: pop-type
-
   pop-member:
-    - match: \b_\b
+    - match: '{{ident_blank}}'
       scope: variable.language.blank.go
-      pop: true
+      pop: 1
     - match: '{{ident}}'
       scope: variable.other.member.go
-      pop: true
+      pop: 1
 
   pop-arguments-starting-with-type:
-    - include: match-comments
     # Go allows functions and types to be wrapped in an arbitrary number of
     # parens. These are not part of the call.
-    - match: \)
-      scope: punctuation.section.parens.end.go
     - match: \(
       scope: punctuation.section.parens.begin.go
-      set:
-        - match: \)
-          scope: punctuation.section.parens.end.go
-          pop: true
-        - include: match-comments
-        - match: (?=\S)
-          set: pop-type
-    - include: pop-on-terminator
+      set: [pop-inside-parens, pop-type]
     - include: pop-before-nonblank
+
+  # Including the newline allows the scope to visually stretch to the right,
+  # and ensures that functionality that relies on comment scoping, such as
+  # contextual hotkeys, works properly at EOL while typing a comment. This
+  # should also match \r\n due to Sublime's internal normalization.
+  pop-on-eol:
+    - match: $\n?
+      pop: 1
+
+  pop-inside-parens:
+    - include: pop-on-paren-end
+    - include: match-any
+
+  match-namespace:
+    - match: '({{ident}})\s*(\.)'
+      captures:
+        1: variable.other.go
+        2: punctuation.accessor.dot.go
+
+  meta-function-declaration:
+    - meta_scope: meta.function.declaration.go
```

Comment:
All those meta-only contexts should contain `- meta_include_prototype: false` to avoid `prototype`s from being prepended. Even if the language itself doesn't define any `prototype` context any inherited syntax might do so in order to inject templating patterns etc.

```suggestion
  meta-function-declaration:
    - meta_include_prototype: false
    - meta_scope: meta.function.declaration.go
```